instagram/src/components/PostContainer/Post.js

```diff
@@ -22,39 +72,39 @@ class Post extends Component {
 
     incrementLikes = (event) => {
         if (!event.target.classList.contains('fas')) {
-            this.setState({
-                likes: this.props.post.likes + 1
+            this.setState((prevState) => {
+                return {
+                    likes: prevState.likes + 1
+                };
             })
         }
         else if (event.target.classList.contains('fas')) {
-            this.setState({
-                likes: this.props.post.likes
+            this.setState((prevState) => {
+                return {
+                    likes: prevState.likes - 1
```

Comment:
This `if...else if` chain looks a little unnecessary. Could you not just use an `if...else` since you're checking the opposite of what the `else if` is checking in your `if`? For example:
```javascript
if (!event.target.classList.contains('fas')) {
  this.setState(prevState => {
    return {
      likes: prevState.likes + 1;
    };
  });
} else {
  this.setState(prevState => {
    return {
      likes: prevState.likes - 1;
    };
  });
}
```
I also see a bit of repetition here. Why not use a variable in your if else instead, and then `setState` afterwards? For example:
```javascript
let amount;
if (!event.target.classList.contains('fas')) {
  amount = 1;
} else {
  amount = -1;
}
this.setState(prevState => {
  return {
    likes: prevState.likes + amount;
  };
});
```
You could even use a ternary now to shorten your code:
```javascript
const { classList } = event.target;
const amount = classList.contains('fas') ? -1 : 1;
this.setState(prevState => {
  return {
    likes: prevState.likes + amount;
  };
});
```